src/source/internal.rs

```diff
@@ -99,335 +88,350 @@ macro_rules! atom_non_zero_telem {
     }
 }
 
+fn handle_internal_metrics(
```

Comment:
This for sure follows the pattern of the original graphite source but that's my fault: a bare function hanging out in the private space of a module is untidy. Each one of these `handle...` functions look to have similar – or the same? – arguments. Can we make `handle` part of the `Source` trait?

Reply:
Untidy is a subjective notion.

We may be able to force `handle` to be in the `Source` trait, however it is going to carry with it the "baggage" of `Send + Sync + Clone` on the state struct for the given source.  For sources this may be ok, as they don't tend to us a lot of external packages to ingest data.

Reply:
It is a subjective notion. I would rather see Source follow the lead of Sink / Filter and be defined in terms of standard interfaces. 

To be clear, the `Send + Sync` constraint exist because we've made the Source responsible for running its own self in a thread where it does not exist for Sink / Filter because bin/cernan.rs runs them in a thread?

Reply:
Right.

I hit this case in https://github.com/postmates/cernan/pull/378 when moving the threading model into the Sink interface.

I will attempt to introduce a `handle` callback to `Source` now and see how much code catches fire.

Reply:
I'll take a look at #378.

Reply:
Most of the action is here - https://github.com/postmates/cernan/pull/378/files#diff-a9892b62ac2d6b57acde6b4ee3d0fb53R46

Reply:
See https://github.com/postmates/cernan/pull/377/commits/344c50ec3cd380dddd60035c6889a063597a8070